Permanent/Modules/Main/Navigation/Routers/SettingsRouter.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  SettingsRouter.swift
+//  Permanent
+//
+//  Created by Lucian Cerbu on 05.02.2024.
+
+import Foundation
+import SwiftUI
+
+class SettingsRouter: ObservableObject {
+    enum Page {
+        case settings
+        case account
+        case storage
+        case myArchives
+        case invitations
+        case activityFeed
+        case security
+        case legacyPlanning
+        case signUp
+    }
+    
+    var rootViewController: RootNavigationController
+    var currentView: Page
+    
+    init(rootViewController: RootNavigationController, currentView: Page) {
+        self.rootViewController = rootViewController
+        self.currentView = currentView
+    }
+    
+    func navigate(to view: Page, router: SettingsRouter) {
+        if view != .settings {
+            self.rootViewController.dismiss(animated: true)
+        }
+        switch view {
+        case .settings:
+            let settingsScreenView = SettingsScreenView(viewModel: StateObject(wrappedValue: SettingsScreenViewModel()), router: router)
+            let host = UIHostingController(rootView: settingsScreenView)
+            self.rootViewController.present(host, animated: true, completion: nil)
+        case .account:
+            let screenView = AccountInfoView()
+            let host = UIHostingController(rootView: screenView)
+            host.modalPresentationStyle = .fullScreen
+            self.rootViewController.present(host, animated: true, completion: nil)
+        case .storage:
+            currentView = .storage
```

Comment:
You only store the currentView for storage. Lower you never set this variable but instead you initialise a let currentView with the view you are presenting. If you use the currentView, you should assign it for every page and you can do that before switch:

currentView = page

Also you should rename the let currentView = InvitesView() to be either invitesView, etc as you did above or rename it everywhere to something generic as rootView.